### PR TITLE
objc.d: don't pass junk bytes to lookup()

### PR DESCRIPTION
**Pull request by WalterBright**
The .size gives the allocated size of the buffer, not the used size.

### CODE REVIEW

File: src/dmd/objc.d
Comment (jacob-carlborg) [inline src/dmd/objc.d]: Why `-1`?

Comment (WalterBright) [reply src/dmd/objc.d]: The terminating 0 is not needed.

Comment (jacob-carlborg) [reply src/dmd/objc.d]: Selectors are 0 terminated.

Comment (WalterBright) [reply src/dmd/objc.d]: Yeah, but the call is to lookup() which then calls stringtable.update() which calls stringtable.update() which calls allocValue() which adds its own 0 termination.

Comment (jacob-carlborg) [reply src/dmd/objc.d]: OK.

**Comment by WalterBright** [comment]
Well, there needs to be an approval before I can merge it.

**Comment by jacob-carlborg** [comment]
I would like to verify that the binary looks correct.

File: src/dmd/objc.d
Comment (jacob-carlborg-derivco) [reply src/dmd/objc.d]: Is there a point in doing line 124 then, if `allocValue` adds its own 0 termination?

**Comment by jacob-carlborg** [comment]
Binary looks good.

File: src/dmd/objc.d
Comment (WalterBright) [reply src/dmd/objc.d]: I'll leave that to you.

Comment (WalterBright) [reply src/dmd/objc.d]: Ok, but I'll leave it to @jacob-carlborg to re-evaluate the logic of the function.

**Review by atilaneves — APPROVED** [review]

**Pull request merged**
